### PR TITLE
Sort player and team names when compiling demo name

### PR DESCRIPTION
**Pull request by osm**

### CODE REVIEW

File: src/g_utils.c [hunk 1]
```diff
@@ -2814,3 +2816,33 @@ char* make_dots(char *dots, size_t dots_len, int cmd_max_len, char *cmd)
...
+			*dst++ = *src;
+		src++;
+        }
+
+	*dst = '\0';
+}
+
+int sort_alphanumeric(const void *a, const void *b)
+{
+	char *aa = strdup(a);
```
Comment (qqshka) [inline src/g_utils.c:2836]: You can’t use heap memory allocation in the mod, use stack, not to mention you doesn’t check success here.

File: src/match.c [hunk 2]
```diff
@@ -2084,8 +2087,15 @@ char* CompilateDemoName()
 				continue;
 			}
 
+			strncpy(players[pc++], name, CLIENT_NAME_LEN);
```
Comment (qqshka) [inline src/match.c:2090]: Replace strncpy with strlcpy

**Review by qqshka — CHANGES_REQUESTED** [review]

File: src/g_utils.c (hunk 1 above)
Comment (osm) [reply src/g_utils.c:2836]: Aha, I wasn't aware that I couldn't use the heap. I'll fix that! (Sorry, it was very sloppy of me to not check if the allocation succeeded)

File: src/match.c (hunk 2 above)
Comment (osm) [reply src/match.c:2090]: Will fix

**Pull request closed without merging**